### DISCUSSION TITLE
[3.x]: Craft is not working when Redis server is down

### DISCUSSION THREAD

**Discussion by martinhellwagner** [craftcms/cms#11360] [Q&A] [answered]
### Description

We have the problem that Craft is not working when the Redis server is down. Is it possible that Craft is using the DB for caching as fallback when the Redis server is down?

### Steps to reproduce

1. Redis server is down
2. Craft is not working

### Expected behavior

Craft should use the DB for caching when the Redis server is down.

### Actual behavior

Craft is not working.

### Craft CMS version

3.7.27

### PHP version

8.0

**Comment by brandonkelly** [comment, marked as answer]
You could accomplish this by setting your `cache` component config to a callback function, which checks whether Redis is running, and if not, fallback to file (or DB) caching.

```php
return [
    'components' => [
        'cache' => function() {
            try {
                Craft::$app->redis->open();
                $useRedis = true;
            } catch (yii\db\Exception $e) {
                $useRedis = false;
            }

            if ($useRedis) {
                $config = [
                    'class' => yii\redis\Cache::class,
                    'defaultDuration' => 86400,
                    'keyPrefix' => craft\helpers\App::env('APP_ID') ?: 'CraftCMS',
                ];
            } else {
                $config = craft\helpers\App::cacheConfig();
            }
            
            return Craft::createObject($config);
        },
    ],
];
```

**Reply by martinhellwagner** [reply]
That makes sense. I'll give it a try. Thank you! 🙂
